src/wallet/rpcdump.cpp

```diff
@@ -419,7 +445,124 @@ Value dumpwallet(const Array& params, bool fHelp)
         }
     }
     file << "\n";
+
+    // dump the zkeys
+    std::set<libzcash::PaymentAddress> addresses;
+    pwalletMain->GetPaymentAddresses(addresses);
+    file << "\n";
+    file << "# Zkeys\n";
+    file << "\n";
+    for (auto addr : addresses ) {
+        libzcash::SpendingKey key;
+        if (pwalletMain->GetSpendingKey(addr, key)) {
+            std::string strTime = EncodeDumpTime(pwalletMain->mapZKeyMetadata[addr].nCreateTime);
+            file << strprintf("%s %s # zaddr=%s\n", CZCSpendingKey(key).ToString(), strTime, CZCPaymentAddress(addr).ToString());
+        }
+    }
+    file << "\n";
+
     file << "# End of dump\n";
     file.close();
     return Value::null;
 }
+
+
+Value z_importkey(const Array& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return Value::null;
+
+    if (fHelp || params.size() < 1 || params.size() > 3)
+        throw runtime_error(
+            "z_importkey \"zkey\" ( \"label\" rescan )\n"
+            "\nAdds a zkey (as returned by z_exportkey) to your wallet.\n"
+            "\nArguments:\n"
+            "1. \"zkey\"             (string, required) The zkey (see z_exportkey)\n"
+            "2. rescan             (boolean, optional, default=true) Rescan the wallet for transactions\n"
+            "\nNote: This call can take minutes to complete if rescan is true.\n"
+            "\nExamples:\n"
+            "\nExport a zkey\n"
+            + HelpExampleCli("z_exportkey", "\"myaddress\"") +
+            "\nImport the zkey with rescan\n"
+            + HelpExampleCli("z_importkey", "\"mykey\"") +
+            "\nImport using a label and without rescan\n"
+            + HelpExampleCli("z_importkey", "\"mykey\" \"testing\" false") +
+            "\nAs a JSON-RPC call\n"
+            + HelpExampleRpc("z_importkey", "\"mykey\", \"testing\", false")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    EnsureWalletIsUnlocked();
+
+    // Whether to perform rescan after import
+    bool fRescan = true;
+    if (params.size() > 1)
+        fRescan = params[1].get_bool();
+
+    string strSecret = params[0].get_str();
+    CZCSpendingKey spendingkey(strSecret);
+    auto key = spendingkey.Get();
+    auto addr = key.address();
+
+    {
+        pwalletMain->MarkDirty();
+
+        // Don't throw error in case a key is already there
+        if (pwalletMain->HaveSpendingKey(addr))
```

Comment:
Fixed now.  Was inherited from upstream method importprivkey as they set a label in the addressbook before checking for a duplicate key.